src/backend/distributed/connection/connection_management.c

```diff
@@ -1448,7 +1448,28 @@ ShouldShutdownConnection(MultiConnection *connection, const int cachedConnection
 bool
 IsCitusInitiatedRemoteBackend(void)
 {
-	return application_name && strcmp(application_name, CITUS_APPLICATION_NAME) == 0;
+	/* does application name exist */
+	if (!application_name)
+	{
+		return false;
+	}
+
+	uint64 prefixLength = strlen(CITUS_APPLICATION_NAME_PREFIX); 
+
+	/* does application name start with Citus's application name prefix */
+	if (strncmp(application_name, CITUS_APPLICATION_NAME_PREFIX, prefixLength) != 0)
```

Comment:
can we use `ExtractGlobalPID` here? I mean, change `ExtractGlobalPID` to get a `char *` and say return `NULL` if cannot be extracted.

And, if returns NULL we return false here?

Reply:
I'm not loving this idea. I would think it'd be the other way around.
Like calling this function (`IsCitusInitiatedRemoteBackend`) before (or in) `ExtractGlobalPID`.
If you insist on this can you explain your reasoning behind it?

Reply:
The idea is to follow Marco's earlier comment, have a single place where we know how to `ExtractGlobalPID` from a given string. Now, we have two, one in `ExtractGlobalPID` the other here. 

> Like calling this function (IsCitusInitiatedRemoteBackend) before (or in) ExtractGlobalPID.

This sounds counterintuitive. Why a utility function like `ExtractGlobalPID` should depend to a lot more generic function `IsCitusInitiatedRemoteBackend`?   

To be able to follow my suggestion, `ExtractGlobalPID` should be changed to a sole utility function, which takes a `char *` and returns globalPID (if not found returns 0/NULL).

Reply:
We had offline discussion and decided to continue as @onderkalaci suggested